### PR TITLE
Use tomllib on Python 3.11+

### PR DESCRIPTION
**Pull request by felixonmars**
Fallback to tomli for compatibility with older Python

### CODE REVIEW

**Comment by Spitfire1900** [comment]
Would you like to do similar for `importlib-metadata` as part of this PR? The PyPi package is only required on Python 3.7.

File: yapf/yapflib/file_resources.py [hunk 1]
```diff
@@ -52,11 +52,14 @@ def _GetExcludePatternsFromPyprojectToml(filename):
   """Get a list of file patterns to ignore from pyproject.toml."""
   ignore_patterns = []
   try:
-    import tomli as tomllib
```
Comment (Spitfire1900) [inline yapf/yapflib/file_resources.py:55]: Not a fan of nested try/catches in favor of https://peps.python.org/pep-0484/#version-and-platform-checking style approach.

I'm also wondering if moving the import to ./yapf/yapflib/__init__.py is a good idea idea vs. conditional login in every file.

Lastly, this package is now part of the package requirements, so managing an `ImportError` may no longer be applicable.

```python
import sys
if sys.version_info >= (3, 10):
    import tomllib
else:
    import tomli as tomllib
```

if ImportError must still be handled.
```python
import sys
if sys.version_info >= (3, 10):
    import tomllib
else:
    try:
        import tomli as tomllib
    except ImportError:
        raise errors.YapfError(
            'tomli package is needed for using pyproject.toml as a '
            'configuration file')
```

**Review by Spitfire1900 — CHANGES_REQUESTED** [review]
Use sys.version_info instead of nested try/catches.

Strip all import try/catches as tomli / tomllib is guaranteed by requirements specification.

**Comment by felixonmars** [comment]
Looks like requested changes are in #1158 now. Closing in favor of that one.

**Pull request closed without merging**